File: src/IEManageSystem.ApiAuthorization/DomainModel/ApiScopes/UserScopeAccessAuthority.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace IEManageSystem.ApiAuthorization.DomainModel.ApiScopes
{
    public class UserScopeAccessAuthority
    {
        public UserScopeAccessAuthority(string scopeName, bool manageAuthority, bool queryAuthority)
        {
            ScopeName = scopeName;

            ManageAuthority = manageAuthority;

            QueryAuthority = queryAuthority;
        }

        public string ScopeName { get; set; }

        public bool ManageAuthority { get; private set; }

        public bool QueryAuthority { get; private set; }
    }
}

File: src/IEManageSystem.Web/ClientApp/src/Common/Core/ApiScopeAuthority/UserScopeAccessAuthority.ts
// 用户对该域的访问权限
export default interface UserScopeAccessAuthority{
    // 域名称
    scopeName: string,
    // 是否拥有该域的管理权限
    manageAuthority: boolean,
    // 是否拥有该域的访问权限
    queryAuthority: boolean
}